.gitlab/pipeline_on_merge.yml

```diff
@@ -1,20 +1,43 @@
-# Copyright (c) 2024 ETH Zurich
-#
-# SPDX-License-Identifier: BSL-1.0
-# Distributed under the Boost Software License, Version 1.0. (See accompanying
-# file BOOST_LICENSE_1_0.rst or copy at http://www.boost.org/LICENSE_1_0.txt)
-
 include:
+  - local: '.gitlab/includes/skip_job.yml'
+    rules:
+      - if: $CI_COMMIT_BRANCH !~ /^gh-readonly-queue/
```

Comment:
_If_ we end up needing the dummy job (see other comment) then make sure this is the complement of the filter in the other jobs.